File: src/contest/w86/t4/qy9Mglee255.cpp
/**
 * // This is the Master's API interface.
 * // You should not implement it, or speculate about its implementation
 * class Master {
 *   public:
 *     int guess(string word);
 * };
 */
class Solution {
public:
    void findSecretWord(vector<string>& words, Master& master) {
        random_device rd;
        mt19937 g(rd());
        auto ss = words;
        while (1) {
            shuffle(ss.begin(), ss.end(), g);
            auto id = g() % ss.size();
            auto get = master.guess(ss[id]);
            if (get == 6) {
                break;
            }
            vector<string> ne;
            for (auto &w : ss) {
                int c = 0;
                for (int i = 0; i < 6; i++) {
                    if (w[i] == ss[id][i]) c++;
                }
                if (c == get) ne.push_back(w);
            }
            ss = ne;
        }
    }
};

File: src/contest/w86/t4/qy9Mglee255.rs
/**
 * // This is the Master's API interface.
 * // You should not implement it, or speculate about its implementation
 * struct Master;
 * impl Master {
 *     fn guess(word:String)->int;
 * };
 */

 impl Solution {
    pub fn find_secret_word(words: Vec<String>, master: &Master) {
        use rand::random;
        let mut ws = words;
        loop {
            for i in (0..ws.len()).rev() {
                ws.swap(i, random::<usize>() % (i + 1))
            }
            let id = random::<usize>() % ws.len();
            let get = master.guess(ws[id].clone());
            if get == 6 {
                break;
            }
            let mut ne = Vec::new();
            for w in ws.clone() {
                let c: i32 = (0..6).map(|x| {
                    if w.chars().nth(x).unwrap() == ws[id].chars().nth(x).unwrap() {
                        1
                    } else {
                        0
                    }
                }).sum();
                if c == get {
                    ne.push(w);
                }
            }
            // println!("{:?}", ne);
            ws = ne.clone()
        }
    }
}